File: ex8misc.py
# ex8: Misc., MATLAB

import numpy as np

foo = True    # Boolean value
bar = None    # Empty value
baz = 'abc'   # String

# Exception handling
try:

    # Expressions
    if foo or (bar is None and baz == 'abc'):
        print('Yep')
    else:
        print('Nope')

    # Multiple return values
    foo, bar = np.meshgrid([1, 2], [1, 2, 3])
    foo = np.meshgrid([1, 2], [1, 2, 3])  # only 1st
    _, foo = np.meshgrid([1, 2], [1, 2, 3])  # only 2nd

except Exception as e:

    # Formatted output
    print(f'Error: {e}')

File: ex8misc.m
% ex8: Misc., MATLAB

foo = 1;      % Boolean value (alt: "true")
bar = [];     % Empty value
baz = 'abc';  % Character array (alt: "abc", string)

% Exception handling
try

    % Expressions
    if foo || (isempty(bar) && strcmp(baz, 'abc'))   % alt: baz == "abc"
        disp('Yep');
    else
        disp('Nope');
    end

    % Multiple return values
    [foo, bar] = meshgrid([1 2], [1 2 3]);
    foo = meshgrid([1 2], [1 2 3]);  % only 1st
    [~, foo] = meshgrid([1 2], [1 2 3]);  % only 2nd

catch ME

    % Formatted output
    fprintf('Error: %s\n', ME.identifier);
end
